snuba/query/parser/__init__.py

```diff
@@ -305,6 +331,38 @@ def _expand_aliases(query: Query) -> None:
     query.transform(visitor)
 
 
+ARRAYJOIN_FUNCTION_MATCH = FunctionCallMatch(None, StringMatch("arrayJoin"), None)
+
+
+def _validate_arrayjoin(query: Query) -> None:
+    # TODO: Actually validate arrayjoin. For now log how it is used.
+    body_arrayjoin = ""
+    arrayjoin = query.get_arrayjoin_from_ast()
+    if arrayjoin is not None:
+        if isinstance(arrayjoin, Column):
+            body_arrayjoin = arrayjoin.column_name
+
+    array_joins = set()
+    for exp in query.get_all_expressions():
+        match = ARRAYJOIN_FUNCTION_MATCH.match(exp)
+        if match is not None:
+            if isinstance(exp, Column):
+                array_joins.add(exp.column_name)
+            elif isinstance(exp, Literal):
+                array_joins.add(f"{exp.value}")
+            else:
+                array_joins.add(f"{exp}")
```

Comment:
You will likely need to limit the size of this. I don't know how our metrics system deals with very long tag values.
Also same here. This can include literals. It is unlikely enough that I would just write the type of expression.